File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {

        Stack<Integer> stack = new Stack<>();

        for(String token : tokens){
            if(token.length() == 1 && token.charAt(0) - '0' < 0){
                int rightNum = stack.pop();
                int leftNum = stack.pop();                
                switch(token){
                    case "+":
                        stack.push(leftNum+rightNum);
                    break;
                    case "-":
                        stack.push(leftNum-rightNum);
                    break;
                    case "*":
                        stack.push(leftNum*rightNum);                    
                    break;
                    case "/":
                        stack.push(leftNum/rightNum);                    
                    break;                                                            
                }
            } else {
                stack.push(Integer.parseInt(token));
            }
        }
        return stack.pop();        
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:

        stack = []        

        for token in tokens:
            if token in "+-*/":
                right = stack.pop()
                left = stack.pop()
                if token == "+":
                    stack.append(left+right)
                elif token == "-":
                    stack.append(left-right)
                elif token == "*":
                    stack.append(left*right)
                elif token == "/":
                     stack.append(int(left/right)) 
            else:
                stack.append(int(token))
        return stack[-1]                               